Add the Samsung S3C2442B CPU idcode to the samsung s3c24xx platform code
and fix a Kconfig typo related to the 2442.

Index: linux-2.6.32.10/arch/arm/mach-s3c2442/Kconfig
===================================================================
--- linux-2.6.32.10.orig/arch/arm/mach-s3c2442/Kconfig
+++ linux-2.6.32.10/arch/arm/mach-s3c2442/Kconfig
@@ -6,7 +6,7 @@
 
 config CPU_S3C2442
 	bool
-	depends on ARCH_S3C2410
+	depends on ARCH_S3C2440
 	select CPU_ARM920T
 	select S3C2410_CLOCK
 	select S3C2410_GPIO
